core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -351,7 +351,16 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   def getAppId: String = get("spark.app.id")
 
   /** Does the configuration contain a given parameter? */
-  def contains(key: String): Boolean = settings.containsKey(key)
+  def contains(key: String): Boolean = {
+    if (settings.containsKey(key)) {
+      true
+    } else {
+      // try to find the settings in the alternatives
+      configsWithAlternatives.get(key).flatMap { alts =>
```

Comment:
this is pretty complicated. Is this not just `configsWithAlternatives.get(key).find(contains(_.key)).isDefined` ? or something 90% like that.